src/builders/ios/index.ts

```diff
@@ -51,6 +52,7 @@ async function initBuilder(ctx: IContext) {
     await fs.ensureDir(dir);
     await fs.chmod(dir, 0o755);
   }
+  await spawnAsync('sudo', ['xcrun', 'simctl', 'list']);
```

Comment:
Was this meant for debugging?

Reply:
Nope, this is a solution for the issue with Xcode 12.1 having problems with detecting available simulators on dry run (and these are apparently needed to complete the build) 🤔
Once this command is invoked before the actual build everything is ok. It's proposed by @wkozyra95 and mirrored from the `turtle v2` repo: https://github.com/expo/turtle-v2/pull/309

<details>
<summary>Without this we get the following error</summary>

```bash
2020-11-03 04:00:55.826 ibtoold[1453:8999] [MT] IBPlatformTool: *** Failed to launch tool with description <IBCocoaTouchPlatformToolDescription: 0x7fe418ac16d0> System content for IBCocoaTouchFramework-fourteenAndLater <IBScaleFactorDeviceTypeDescription: 0x7fe4187ba8e0> scaleFactor=2x, renderMode.identifier=(null): Failed to find or create execution context for description '<IBCocoaTouchPlatformToolDescription: 0x7fe418ac16d0> System content for IBCocoaTouchFramework-fourteenAndLater <IBScaleFactorDeviceTypeDescription: 0x7fe4187ba8e0> scaleFactor=2x, renderMode.identifier=(null)'.
Device type: IBSimDeviceTypeiPad2x (com.apple.dt.Xcode.IBSimDeviceType.iPad-2x)
Sim runtime: (null)
** Please also include the output of `xcrun simctl list` and `xcode-select -p`.: Failed to find newest available Simulator runtime: No available runtime found from: [
    watchOS 7.0 (7.0 - 18R382) - com.apple.CoreSimulator.SimRuntime.watchOS-7-0 (unavailable, failed to open liblaunch_sim.dylib) ==> not available: Error Domain=com.apple.CoreSimulator.SimError Code=401 "The watchOS 7.0 simulator runtime is not available." UserInfo={NSLocalizedDescription=The watchOS 7.0 simulator runtime is not available., NSUnderlyingError=0x7fe418930b20 {Error Domain=NSPOSIXErrorDomain Code=53 "Software caused connection abort" UserInfo={NSLocalizedFailureReason=XPC error talking to SimLaunchHostService: <error: 0x7fff97b3e9a0> { count = 1, transaction: 0, voucher = 0x0, contents =
	"XPCErrorDescription" => <string: 0x7fff97b3eb08> { length = 18, contents = "Connection invalid" }
}}}, NSLocalizedRecoverySuggestion=Unable to open liblaunch_sim.dylib.  Try reinstalling Xcode or the simulator runtime., NSLocalizedFailureReason=failed to open liblaunch_sim.dylib},
    tvOS 14.0 (14.0 - 18J383) - com.apple.CoreSimulator.SimRuntime.tvOS-14-0 (unavailable, failed to open liblaunch_sim.dylib) ==> not available: Error Domain=com.apple.CoreSimulator.SimError Code=401 "The tvOS 14.0 simulator runtime is not available." UserInfo={NSLocalizedDescription=The tvOS 14.0 simulator runtime is not available., NSUnderlyingError=0x7fe41893aea0 {Error Domain=NSPOSIXErrorDomain Code=53 "Software caused connection abort" UserInfo={NSLocalizedFailureReason=XPC error talking to SimLaunchHostService: <error: 0x7fff97b3e9a0> { count = 1, transaction: 0, voucher = 0x0, contents =
	"XPCErrorDescription" => <string: 0x7fff97b3eb08> { length = 18, contents = "Connection invalid" }
}}}, NSLocalizedRecoverySuggestion=Unable to open liblaunch_sim.dylib.  Try reinstalling Xcode or the simulator runtime., NSLocalizedFailureReason=failed to open liblaunch_sim.dylib},
    iOS 14.1 (14.1 - 18A8394) - com.apple.CoreSimulator.SimRuntime.iOS-14-1 (unavailable, failed to open liblaunch_sim.dylib) ==> not available: Error Domain=com.apple.CoreSimulator.SimError Code=401 "The iOS 14.1 simulator runtime is not available." UserInfo={NSLocalizedDescription=The iOS 14.1 simulator runtime is not available., NSUnderlyingError=0x7fe418930c60 {Error Domain=NSPOSIXErrorDomain Code=53 "Software caused connection abort" UserInfo={NSLocalizedFailureReason=XPC error talking to SimLaunchHostService: <error: 0x7fff97b3e9a0> { count = 1, transaction: 0, voucher = 0x0, contents =
	"XPCErrorDescription" => <string: 0x7fff97b3eb08> { length = 18, contents = "Connection invalid" }
}}}, NSLocalizedRecoverySuggestion=Unable to open liblaunch_sim.dylib.  Try reinstalling Xcode or the simulator runtime., NSLocalizedFailureReason=failed to open liblaunch_sim.dylib},
] when matching for <IBCocoaTouchPlatformToolDescription: 0x7fe418ac16d0> System content for IBCocoaTouchFramework-fourteenAndLater <IBScaleFactorDeviceTypeDescription: 0x7fe4187ba8e0> scaleFactor=2x, renderMode.identifier=(null), idiom=<IBIPhoneIdiom: 0x7fe4187ac2a0> runtime=<IBCocoaTouchTargetRuntime: 0x7fe4180d25e0>
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>com.apple.ibtool.errors</key>
	<array>
		<dict>
			<key>description</key>
			<string>Failed to find or create execution context for description '&lt;IBCocoaTouchPlatformToolDescription: 0x7fe418ac16d0&gt; System content for IBCocoaTouchFramework-fourteenAndLater &lt;IBScaleFactorDeviceTypeDescription: 0x7fe4187ba8e0&gt; scaleFactor=2x, renderMode.identifier=(null)'.
Device type: IBSimDeviceTypeiPad2x (com.apple.dt.Xcode.IBSimDeviceType.iPad-2x)
Sim runtime: (null)
** Please also include the output of `xcrun simctl list` and `xcode-select -p`.</string>
			<key>underlying-errors</key>
			<array>
				<dict>
					<key>description</key>
					<string>Failed to find newest available Simulator runtime</string>
					<key>underlying-errors</key>
					<array>
						<dict>
							<key>description</key>
							<string>No available runtime found from: [
    watchOS 7.0 (7.0 - 18R382) - com.apple.CoreSimulator.SimRuntime.watchOS-7-0 (unavailable, failed to open liblaunch_sim.dylib) ==&gt; not available: Error Domain=com.apple.CoreSimulator.SimError Code=401 "The watchOS 7.0 simulator runtime is not available." UserInfo={NSLocalizedDescription=The watchOS 7.0 simulator runtime is not available., NSUnderlyingError=0x7fe418930b20 {Error Domain=NSPOSIXErrorDomain Code=53 "Software caused connection abort" UserInfo={NSLocalizedFailureReason=XPC error talking to SimLaunchHostService: &lt;error: 0x7fff97b3e9a0&gt; { count = 1, transaction: 0, voucher = 0x0, contents =
	"XPCErrorDescription" =&gt; &lt;string: 0x7fff97b3eb08&gt; { length = 18, contents = "Connection invalid" }
}}}, NSLocalizedRecoverySuggestion=Unable to open liblaunch_sim.dylib.  Try reinstalling Xcode or the simulator runtime., NSLocalizedFailureReason=failed to open liblaunch_sim.dylib},
    tvOS 14.0 (14.0 - 18J383) - com.apple.CoreSimulator.SimRuntime.tvOS-14-0 (unavailable, failed to open liblaunch_sim.dylib) ==&gt; not available: Error Domain=com.apple.CoreSimulator.SimError Code=401 "The tvOS 14.0 simulator runtime is not available." UserInfo={NSLocalizedDescription=The tvOS 14.0 simulator runtime is not available., NSUnderlyingError=0x7fe41893aea0 {Error Domain=NSPOSIXErrorDomain Code=53 "Software caused connection abort" UserInfo={NSLocalizedFailureReason=XPC error talking to SimLaunchHostService: &lt;error: 0x7fff97b3e9a0&gt; { count = 1, transaction: 0, voucher = 0x0, contents =
	"XPCErrorDescription" =&gt; &lt;string: 0x7fff97b3eb08&gt; { length = 18, contents = "Connection invalid" }
}}}, NSLocalizedRecoverySuggestion=Unable to open liblaunch_sim.dylib.  Try reinstalling Xcode or the simulator runtime., NSLocalizedFailureReason=failed to open liblaunch_sim.dylib},
    iOS 14.1 (14.1 - 18A8394) - com.apple.CoreSimulator.SimRuntime.iOS-14-1 (unavailable, failed to open liblaunch_sim.dylib) ==&gt; not available: Error Domain=com.apple.CoreSimulator.SimError Code=401 "The iOS 14.1 simulator runtime is not available." UserInfo={NSLocalizedDescription=The iOS 14.1 simulator runtime is not available., NSUnderlyingError=0x7fe418930c60 {Error Domain=NSPOSIXErrorDomain Code=53 "Software caused connection abort" UserInfo={NSLocalizedFailureReason=XPC error talking to SimLaunchHostService: &lt;error: 0x7fff97b3e9a0&gt; { count = 1, transaction: 0, voucher = 0x0, contents =
	"XPCErrorDescription" =&gt; &lt;string: 0x7fff97b3eb08&gt; { length = 18, contents = "Connection invalid" }
}}}, NSLocalizedRecoverySuggestion=Unable to open liblaunch_sim.dylib.  Try reinstalling Xcode or the simulator runtime., NSLocalizedFailureReason=failed to open liblaunch_sim.dylib},
] when matching for &lt;IBCocoaTouchPlatformToolDescription: 0x7fe418ac16d0&gt; System content for IBCocoaTouchFramework-fourteenAndLater &lt;IBScaleFactorDeviceTypeDescription: 0x7fe4187ba8e0&gt; scaleFactor=2x, renderMode.identifier=(null), idiom=&lt;IBIPhoneIdiom: 0x7fe4187ac2a0&gt; runtime=&lt;IBCocoaTouchTargetRuntime: 0x7fe4180d25e0&gt;</string>
						</dict>
					</array>
				</dict>
			</array>
		</dict>
	</array>
</dict>
</plist>
```
</details>

Reply:
Does this work without sudo, which would be preferable?

Reply:
I don't remember for sure, but I think I tested that and it didn't work. I also tried to do that on VM template directly, but it's necessary after every restart.

Not really sure what is the reason, the issue only appears when starting jobs via govc , if the build is started via ssh everything works. This was always happening, but before simulator runtime was not used in builds.

Use of sudo here shouldn't be a problem, on turtle-v1 there is no way to execute any user code and on turtle-v2 we are running it as apart of VM setup, so no user code is present at that point.